src/include/network/services/host.rb

```diff
@@ -175,36 +171,30 @@ def HostsMainDialog(standalone)
       end
 
       UI.ChangeWidget(Id(:table), :Items, table_items)
-      UI.SetFocus(Id(:table)) if Ops.greater_than(Builtins.size(table_items), 0)
+      UI.SetFocus(Id(:table)) if !table_items.empty?
```

Comment:
Better to use `any?`

```suggestion
      UI.SetFocus(Id(:table)) if table_items.any?
```

Reply:
I must admit that I say "better to use" meaning "better to read". That said, if you know more about the performance, there is a [bit old discussion about it](https://github.com/JuanitoFatas/fast-ruby/pull/49) in the https://github.com/JuanitoFatas/fast-ruby repo .

Below, you can find three [`compare!`](https://www.rubydoc.info/gems/benchmark-ips/Benchmark/Compare) samples from three different execution of a slightly modified [benchmark of the next-to-last comment](https://github.com/JuanitoFatas/fast-ruby/pull/49#issuecomment-128444354) (to use `ary.any?` instead of `ary.any? { true }`), using `ruby 2.5.3p105`

```
Comparison:
         !ary.empty?:  3408896.0 i/s
       ary.size != 0:  3294235.0 i/s - same-ish: difference falls within error
            ary.any?:  2937220.1 i/s - same-ish: difference falls within error
     !ary.size.zero?:  2871812.4 i/s - same-ish: difference falls within error
           ary != []:  2005702.0 i/s - 1.70x  slower
 !!ary.size.nonzero?:  1305874.0 i/s - 2.61x  slower
```

```
Comparison:
         !ary.empty?:  3346054.4 i/s
       ary.size != 0:  3161546.0 i/s - same-ish: difference falls within error
            ary.any?:  2999996.7 i/s - 1.12x  slower
     !ary.size.zero?:  2823247.6 i/s - 1.19x  slower
           ary != []:  1974117.8 i/s - 1.69x  slower
 !!ary.size.nonzero?:  1278290.9 i/s - 2.62x  slower

```

```
Comparison:
         !ary.empty?:  3366606.0 i/s
       ary.size != 0:  3169765.1 i/s - same-ish: difference falls within error
            ary.any?:  3068372.1 i/s - same-ish: difference falls within error
     !ary.size.zero?:  2870325.6 i/s - 1.17x  slower
           ary != []:  2124715.2 i/s - 1.58x  slower
 !!ary.size.nonzero?:  1350402.9 i/s - 2.49x  slower
```

I hope this information could be useful to help you make a decision.

Reply:
huh. very exhaustive, thank you.